src/networking.c

```diff
@@ -319,6 +319,46 @@ int prepareClientToWrite(client *c) {
     return C_OK;
 }
 
+/* Returns everything in the client reply linked list in a SDS format.
+ * This should only be used only with a caching client. */
```

Comment:
> This should only be used only with a caching client.

Is there a way to assert this?

Reply:
We have added an assertion below to confirm client's static buffer is not used beyond that it should be fine to use this method. Introducing another struct seems quite some work for this usecase.